utils/form_validation.php

```diff
@@ -264,7 +332,7 @@ class DateField extends Field
 {
     function check(string $msg = null)
     {
-        if (!preg_match("/^[\d-]*$/", $this->value)) {
+        if (!preg_match("/^[\d-]*$/", $this->value ?? "")) {
```

Comment:
Look into turning this generic line into a private method inside `Field`, like this:
```php
private function test($preg){
  return preg_match($preg, $this->value ?? "");
}
```